### PR TITLE
MULE-9556 Support non-blocking processing in one-way flows

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: core/src/main/java/org/mule/DefaultMuleMessage.java [hunk 1]
```diff
@@ -2111,7 +2111,6 @@ protected void copyToInbound(DefaultMuleMessage newMessage) throws Exception
         }
 
         newMessage.clearProperties(PropertyScope.INBOUND);
-        newMessage.clearProperties(PropertyScope.INVOCATION);
```
Comment (pablokraan) [inline core/src/main/java/org/mule/DefaultMuleMessage.java]: Why are you removing this line?

File: core/src/main/java/org/mule/construct/Flow.java [hunk 2]
```diff
@@ -155,13 +137,40 @@ public MuleEvent process() throws Exception
     private MuleEvent createMuleEventForCurrentFlow(MuleEvent event, Object replyToDestination, ReplyToHandler
             replyToHandler)
     {
+        if (replyToHandler != null && !(replyToHandler instanceof DefaultReplyToHandler))
```
Comment (pablokraan) [inline core/src/main/java/org/mule/construct/Flow.java]: This instance of seems ugly to me. Why only the DefaultReplyToHandler is a special case?

File: core/src/main/java/org/mule/construct/Flow.java [hunk 3]
```diff
@@ -155,13 +137,40 @@ public MuleEvent process() throws Exception
...
+        {
+            replyToHandler = createNonBlockingReplyToHandler(event, replyToHandler);
+        }
+        else
+        {
+            replyToHandler = null;
```
Comment (pablokraan) [inline core/src/main/java/org/mule/construct/Flow.java]: Why is original replyToHandler nullified here?

File: core/src/main/java/org/mule/DefaultMuleMessage.java (hunk 1 above)
Comment (dfeist) [inline core/src/main/java/org/mule/DefaultMuleMessage.java]: I thought this was incorrect, but I see now it is used in VM and in this use it is correct to remove these.  I'll put this back and instead use a different method in TestMessageDispatcher.

**Comment by pablokraan** [comment]
This PR is for a new feature... but we are on feature freeze mode....

File: core/src/main/java/org/mule/DefaultMuleMessage.java (hunk 1 above)
Comment (dfeist) [inline core/src/main/java/org/mule/DefaultMuleMessage.java]: I thought this was wrong, but now see it's used by VM transport. I'll revert this and create response message differently in TestMessageDispatcher.

File: core/src/main/java/org/mule/construct/Flow.java (hunk 3 above)
Comment (dfeist) [inline core/src/main/java/org/mule/construct/Flow.java]: This is also being done before this PR, just differently.

**Pull request merged**